### PR TITLE
[staging-next] several gcc11 fixes

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
```
logger.C: In function 'void log_sync(logline&)':
logger.C:101:14: warning: ignoring return value of 'ssize_t writev(int, const iovec*, int)' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  101 |       writev (STDERR_FILENO, iov, 2);
      |       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
logger.C:103:16: warning: ignoring return value of 'ssize_t writev(int, const iovec*, int)' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  103 |         writev (logfd, iov, 2);
      |         ~~~~~~~^~~~~~~~~~~~~~~
make[1]: *** [Makefile:431: logger.o] Error 1
make[1]: Leaving directory '/build/deliantra-server-3.1/common'
make: *** [Makefile:390: all-recursive] Error 1
builder for '/nix/store/b89j3hnqx51kwjyvqsy1zmds1c8z03i6-deliantra-server-3.1.drv' failed with exit code 2

**Comment by fabianhjr** [comment]
Thanks for reviewing, refactor deliandra packages to use gcc10 instead of gcc11, seems like there is a bug upstream that wasn't caught before and there hasn't been new releases in a while.

**Comment by mweinelt** [comment]
@ofborg build deliantra-server

File: pkgs/tools/networking/cjdns/default.nix [hunk 1]
```diff
@@ -15,7 +15,14 @@ stdenv.mkDerivation rec {
     # for flock
     lib.optional stdenv.isLinux util-linux;
 
-  CFLAGS = "-O2 -Wno-error=stringop-truncation";
+  NIX_CFLAGS_COMPILE = [
+    "-O2"
+    "-Wno-error=array-bounds"
+    "-Wno-error=stringop-overflow"
+    "-Wno-error=stringop-overread"
+    "-Wno-error=stringop-truncation"
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/cjdns/default.nix:23]: This should be hardeningDisable

**Pull request merged**